scripts/digest_inputs

```diff
@@ -54,18 +67,40 @@ add_vars TAGS REDHAT_TAG DEV_TAGS WORKDIR DOCKERFILE
 
 # Determine repo name values.
 REPO_NAME_MINUS_ENTERPRISE="${REPO_NAME%%-enterprise}"
+
 # Assume this is not enterprise.
 ENTERPRISE_DETECTED=false
-ENT_META=""
+
+#
+## Version mangling.
+#
+# Catch and auto-correct a few common mistakes with the version input.
+# Emit loud warnings when action is taken based on these rules.
+
 # This checks if removing '-enterprise' suffix actually did anything.
 # If it did, then we've detected a likely enterprise repo.
 [ "$REPO_NAME" != "$REPO_NAME_MINUS_ENTERPRISE" ] && {
   ENTERPRISE_DETECTED=true
-  if ! [[ "$VERSION" =~ .*\+.* ]]; then
-	VERSION="$VERSION+ent"
+  # If the version doesn't contain a '+' or '-ent' then add the '+ent' suffix.
+  if [[ ! "$VERSION" =~ .*\+.* ]] && [[ ! "$VERSION" =~ .*-ent.* ]]; then
+	NEW_VERSION="$VERSION+ent"
+    warn "Appending '+ent' to the version string '$VERSION' -> '$NEW_VERSION' add '+ent' to the version input to remove this warning."
+	VERSION="$NEW_VERSION"
   fi
 }
 
+# If the version contains '-ent' then rectify this to '+ent' for the sake
+# getting the correct default zip name for the product.
+if [[ "$VERSION" =~ .*-ent.* ]]; then
+	NEW_VERSION="${VERSION//-ent/+ent}"
+	warn "Changing '-ent' to '+ent' in version string; '$VERSION' -> '$NEW_VERSION'; change the version input to '$NEW_VERSION' to remove this warning."
```

Comment:
Modifying this warning message to state why it's changing would be good. Written as is if someone sees it in logs it  isn't clear why the version is changing and what impact it will have. In this case, this is internal to the action to grab the zip so it won't actually affect any final processed artifact so this would make folks feel better about seeing this message.

Reply:
That's a good point, I'll make the message clearer, thanks.